Gems/ROS2/Code/Source/Utilities/PhysicsCallbackHandler.cpp

```diff
@@ -11,30 +11,18 @@
 
 namespace ROS2::Utils
 {
-    void PhysicsCallbackHandler::InstallPhysicalCallback(AZ::EntityId entityId)
+    void PhysicsCallbackHandler::InstallPhysicalCallback()
```

Comment:
I would propose to make AZ::EntityId entityId optional here to prevent breaking change. 
If the API consumer won't provide entiyId the callback will be installed, but the handle m_bodyHandle will be invalid.

Reply:
I suggest to have only 1 InstallPhysicalCallback which has no arguments.
1. Use RequiredServices in implementing classes first and foremost, if possible, to define requirements for the entity. 
2. If requirements are not simple or static, implement OnPhysicsInitialization to check whether our Entity meets constraints. Allow it to fail and disconnect callback & inform user through error log. You can acquire enityId in implementation body.
3. OnPhysicsInitialization will also cache all bus calls / getters so that it is not repeated in OnPhysicsSimulationFinished.
4. Document accordingly, including how a new sensor component may want to derive the class AND ROS2SensorComponent.
5. Expose PhysicsCallbackHandler as an interface for developers.

Please also correct m_entityId passed as function argument - this is wrong naming convention for a function argument.

If so, adjust IMU / Odometry sensor - overriding OnPhysicsInitialization.

@michalpelka if you have other design in mind lets discuss.

Reply:
@adamdbrw totally agree.
- remove 'AzPhysics::SimulatedBodyHandle m_bodyHandle' from PhysicsCallbackHandler.
- add virtual, empty `PhysicsCallbackHandler::OnPhysicsInitialization` whiteout parameters.
- To implement "fast physical sensor" the Developer need to:
  - derive ROS2SensorComponent
  - derive PhysicsCallbackHandler
  - implement `PhysicsCallbackHandler::OnPhysicsInitialization` where she or he will cache all needed pointers handlers and use them in implementation PhysicsCallbackHandler::OnPhysicsInitialization.
  
I would only propose to add utility function / static member to easily get something like `AZStd::unorderedmap<AZ::EntityId,AzPhysics::SimulatedBodyHandle>` from given Entity eg : ` AZStd::unorderedmap<AZ::EntityId,AzPhysics::SimulatedBodyHandle> getHandlesFromEntity(const AZ::EntityId &id);` - to limit redundant code in `PhysicsCallbackHandler::OnPhysicsInitialization`

Reply:
But currently the ROS2SensorComponent doesn't do much other than providing a mechanism for frame based publication (so it is only really useful for slow sensors). This is related to https://github.com/o3de/o3de-extras/issues/295.
Maybe ROS2SensorComponent should derive from PhysicsCallbackHandler and always use the physics engine updates? Maybe it should just be called ROS2RegularPublisher?

Reply:
Discussed, will implement Michał's solution